File: langstack/langstack-front/domains/response/InitResponse.ts
import Activity from "@/domains/activity/Activity";
import Card from "@/domains/card/Card";
import Genre from "@/domains/genre/Genre";

export default class InitResponse {
    constructor(
        public theLastFiveCards: Card[] = [],
        public theLastFiveDaysActivity: Activity[] = [],
        public allGenres: Genre[] = [],
        public allCardCount: number = 0
    ) {}
}

File: langstack/langstack-back/src/main/java/jp/langstack/interfaces/response/InitResponse.java
package jp.langstack.interfaces.response;

import java.util.List;
import java.util.Set;

import lombok.Builder;
import lombok.Data;

@Builder
@Data
public class InitResponse {
 
    // postDateが最新5つのカード(CardService)
    private List<CardResponse> theLastFiveCards;
    // 今日から5日前までの投稿履歴(CardService)
    private Set<Activity> theLastFiveDaysActivity;
    // OK GenreService
    private List<GenreResponse> allGenres;
    // CardService
    private int allCardCount;

}
